webapi/CopilotChat/Controllers/DocumentImportController.cs

```diff
@@ -126,18 +126,21 @@ private enum SupportedFileType
 
         this._logger.LogInformation("Importing {0} document(s)...", documentImportForm.FormFiles.Count());
 
-        // TODO: [Issue #49] Perform the import in parallel.
         DocumentMessageContent documentMessageContent = new();
         IEnumerable<ImportResult> importResults = new List<ImportResult>();
-        foreach (var formFile in documentImportForm.FormFiles)
-        {
-            var importResult = await this.ImportDocumentHelperAsync(kernel, formFile, documentImportForm);
-            documentMessageContent.AddDocument(
-                formFile.FileName,
-                this.GetReadableByteString(formFile.Length),
-                importResult.IsSuccessful);
-            importResults = importResults.Append(importResult);
-        }
+        await Task.WhenAll(documentImportForm.FormFiles.Select(formFile =>
```

Comment:
We need to make sure the kernel supports inserting memories in parallel. 
My experience with this is that the VolatileMemoryStore is not made to support inserting memories in parallel. Specially, you could potentially end up getting this exception: https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs#L25.

Reply:
this has merged and the exception should no longer be sent. specifically, when `CreateCollectionAsync` is called during the memory write flow, `VolatileMemoryStore` will no longer throw an exception if the collection already exists.

Reply:
we will need to consume the updated nuget to benefit from this change, but this work should happen soon as we have been doing this regularly.